src/Core/Models/Byte.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+namespace Microsoft\Graph\Core\Models;
+
+use InvalidArgumentException;
+use JsonSerializable;
+
+class Byte implements JsonSerializable
+{
+    /**
+     * @var int|null $value
+     */
+    private $value;
+
+    /**
+     * @param int|null $value
+     */
+    public function __construct(?int $value) {
+        if (is_null($value)) {
+            return;
+        }
+        $this->value = $value;
+        if($this->value < 0 || $this->value > 255) {
+            throw new InvalidArgumentException("Byte should be a value between 1-255 inclusive {$value} given");
+        }
+    }
+
+    /**
+     * @return int
+     */
+    public function jsonSerialize(): int {
+        return $this->value;
```

Comment:
Perhaps we should set a default for `$value` as `0` incase it's initialised with null for the current implementation